### PR TITLE
Add Msf::Post::Unix.is_root? method

### PR DESCRIPTION
**Pull request by bcoles**
Add `Msf::Post::Unix.is_root?` and update FreeBSD and OpenBSD local exploit modules to use `Msf::Post::Unix.is_root?`.

### CODE REVIEW

File: lib/msf/core/post/unix.rb [hunk 1]
```diff
@@ -4,6 +4,12 @@
 
 module Msf::Post::Unix
 
+  #
+  # @return [Boolean] true if session is running as uid=0
+  #
+  def is_root?
+    (cmd_exec('id -u').to_s.gsub(/[^\d]/, '') == '0')
```
Comment (adfoster-r7) [inline lib/msf/core/post/unix.rb:11]: Makes sense to me :+1: I'm still unfamiliar with the codebase, what's the semantic difference between this and the similar function below?

https://github.com/rapid7/metasploit-framework/blob/090b80eea74aa812bb8dff933a9817b7f0282145/lib/msf/core/post/linux/priv.rb#L10-L30

Comment (wvu-r7) [reply lib/msf/core/post/unix.rb:11]: Hmm, I thought it looked familiar.

#10119

Comment (bcoles) [reply lib/msf/core/post/unix.rb:11]: This is a UNIX mixin.

Comment (wvu-r7) [reply lib/msf/core/post/unix.rb:11]: Would be nice to unify some of these methods in the future.

Comment (bcoles) [reply lib/msf/core/post/unix.rb:11]: > Would be nice to unify some of these methods in the future.

Unification seems like premature over-engineering. I'd rather see code duplication between mixins than have mixins within a platform identity crisis.

If the functionality is relevant to multiple platforms, then duplicated functionality can later be unified in a single lib. However, given that the unix post library is already having an existential crisis (is it unix, is it osx?) and has been neglected for a decade, along with its osx counterpart, that doesn't bode well.

I'd prefer to see the unix mixin vaporised and replaced with a POSIX mixin for use in other mixins, and a new bsd mixin for various BSDs (openbsd, hardenedbsd, freebsd, netbsd, dragonflybsd, trubsd, ...).

Comment (wvu-r7) [reply lib/msf/core/post/unix.rb:11]: > I'd prefer to see the unix mixin vaporised and replaced with a POSIX mixin for use in other mixins, and a new bsd mixin for various BSDs (openbsd, hardenedbsd, freebsd, netbsd, dragonflybsd, trubsd, ...).

This is all I want to see, tbh. Sounds like you're projecting a bit.

**Comment by bcoles** [comment]
# Release Notes

Add `Msf::Post::Unix.is_root?` method to the UNIX post mixin.

**Review by wvu — COMMENTED** [review]
#13837

**Pull request merged**